src/main/java/com/repository/WalletRepository.java

```diff
@@ -9,12 +9,11 @@
 @Repository
 public interface WalletRepository extends JpaRepository<Wallet, Integer> {
 
-    @Query(value = "SELECT w FROM Wallet w, User u WHERE w.user=u and u.id = ?1")
     List<Wallet> findUserWallets(Integer id);
 
-    @Query(value = "SELECT w FROM Wallet w, User u WHERE w.user=u and u.id = ?1 and w.id=?2")
     Wallet findUsersWalletByID(Integer idUser, Integer idWallet);
 
-    @Query(value = "SELECT w FROM Wallet w WHERE w.id=?1")
     Wallet findWalletByID(Integer idWallet);
+
+    void deleteById(Integer walletId);
```

Comment:
Rethink is you really need delete operation - you will lost historic data, maybe it would be better not to delete but just mark some flag such as deletedByUser as true and do not show records with that flag for users?